### PR TITLE
refactor(Message): accept a single object instead of 3 arguments

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**
This changes the Message#startThread parameters to be consistent with ThreadManager#create to also prevent the need for any breaking changes in the future if extra parameters need to be added.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: src/structures/Message.js [hunk 1]
```diff
@@ -714,15 +714,22 @@ class Message extends Base {
...
    * Create a new public thread from this message
    * @see ThreadManager#create
-   * @param {string} name The name of the new Thread
-   * @param {ThreadAutoArchiveDuration} autoArchiveDuration How long before the thread is automatically archived
-   * @param {string} [reason] Reason for creating the thread
+   * @param {StartThreadOptions} [options] Options for starting a thread on this message
    * @returns {Promise<ThreadChannel>}
    */
-  startThread(name, autoArchiveDuration, reason) {
+  startThread({ name, autoArchiveDuration, reason } = {}) {
```
Comment (NotSugden) [inline src/structures/Message.js:732]: what about 
```suggestion
  startThread(options = {}) {
```
and then below
`create({ ...options, startMessage: this });`

Comment (ImRodry) [reply src/structures/Message.js:732]: Thought of this but didn't know the best way to implement it, thanks

File: typings/index.d.ts [hunk 2]
```diff
@@ -4316,12 +4318,9 @@ export type ThreadChannelResolvable = ThreadChannel | Snowflake;
 
 export type ThreadChannelTypes = 'GUILD_NEWS_THREAD' | 'GUILD_PUBLIC_THREAD' | 'GUILD_PRIVATE_THREAD';
 
-export interface ThreadCreateOptions<AllowedThreadType> {
-  name: string;
-  autoArchiveDuration: ThreadAutoArchiveDuration;
+export interface ThreadCreateOptions<AllowedThreadType> extends StartThreadOptions {
```
Comment (shinotheshino) [inline typings/index.d.ts:4321]: Shouldn't this be:
```suggestion
export interface ThreadCreateOptions<AllowedThreadType extends ThreadChannelTypes> extends StartThreadOptions {
```
?

**Review by kyranet — APPROVED** [review]

File: typings/index.d.ts (hunk 2 above)
Comment (ImRodry) [reply typings/index.d.ts]: No becauseThreadChannelTypes contains types that AllowedThreadTypes doesn’t

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**